tests/fsharp/Compiler/ErrorMessages/SuggestionsTests.fs

```diff
@@ -0,0 +1,345 @@
+        CompilerAssert.TypeCheckSingleError
+            """
+type MyClass() =
+    member val MyProperty = "" with get, set 
+    member val MyProperty2 = "" with get, set
+    member val ABigProperty = "" with get, set
+
+let c = MyClass(Property = "")
+            """
+            FSharpErrorSeverity.Error
+            495
+            (7, 17, 7, 25)
+            "The object constructor 'MyClass' has no argument or settable return property 'Property'. The required signature is new : unit -> MyClass."
+
+
+    [<Test>]
+    let ``Suggest Generic Type`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+type T = System.Collections.Generic.Dictionary<int11,int>
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 48, 2, 53)
+            "The type 'int11' is not defined."
+
+
+    [<Test>]
+    let ``Suggest Methods`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test2 = 
+    type D() = 
+
+       static let x = 1
+
+       member x.Method1() = 10
+
+    D.Method2()
+            """
+            FSharpErrorSeverity.Error
+            39
+            (9, 7, 9, 14)
+            "The field, constructor or member 'Method2' is not defined."
+
+
+    [<Test>]
+    let ``Suggest Modules`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Collections =
+
+    let f () = printfn "%s" "Hello"
+
+open Collectons
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 6, 6, 16)
+            "The namespace or module 'Collectons' is not defined."
+
+
+    [<Test>]
+    let ``Suggest Namespaces`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+open System.Collectons
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 13, 2, 23)
+            "The namespace 'Collectons' is not defined."
+
+
+    //TODO This was marked with SCFLAGS="--vserrors" in tests/fsharpqa/Source/Warnings/env.lst file
+    [<Test>]
+    let ``Suggest Record Labels`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+type MyRecord = { Hello: int; World: bool}
+
+let r = { Hello = 2 ; World = true}
+
+let x = r.ello
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 11, 6, 15)
+            "The field, constructor or member 'ello' is not defined."
+
+
+    //TODO This was marked with SCFLAGS="--vserrors" in tests/fsharpqa/Source/Warnings/env.lst file
+    [<Test>]
+    let ``Suggest Record Type for RequireQualifiedAccess Records`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+[<RequireQualifiedAccess>]
+type MyRecord = {
+    Field1: string
+    Field2: int
+}
+
+let r = { Field1 = "hallo"; Field2 = 1 }
+            """
+            FSharpErrorSeverity.Error
+            39
+            (8, 11, 8, 17)
+            "The record label 'Field1' is not defined."
+
+
+    ///TODO Check if the second error message is correct
```

Comment:
I think this comment can be removed:

We could improve the error message in this situation to something like the previous error, which would be helpful but that is a separate issue.
````
 Did you intend to access the indexer via d.[index] instead?````
````
This output is generated by the RTM compiler
```
C:\temp>fsc foo3.fs
Microsoft (R) F# Compiler version 10.5.0.0 for F# 4.6
Copyright (c) Microsoft Corporation. All Rights Reserved.

foo3.fs(2,9): error FS3217: This value is not a function and cannot be applied. Did you intend to access the indexer via d.[index] instead?

foo3.fs(4,9): error FS0003: This value is not a function and cannot be applied.

foo3.fs(7,10): error FS3217: This expression is not a function and cannot be applied. Did you intend to access the indexer via expr.[index] instead?

C
```